src/node_jsvmapi_types.h

```diff
@@ -34,4 +36,48 @@ struct napi_property_descriptor {
   void* data;
 };
 
+enum napi_valuetype {
+  // ES6 types (corresponds to typeof)
+  napi_undefined,
+  napi_null,
+  napi_boolean,
+  napi_number,
+  napi_string,
+  napi_symbol,
+  napi_object,
+  napi_function,
+};
+
+enum napi_typedarray_type {
```

Comment:
Yeah. May be worth thinking about exposing some of those V8 methods via NAPI wrappers to avoid problems like this. Instead of using napi_get_type_of_value to check for napi_object, just use IsObject or IsTypedArray or IsUint8Array directly. Shouldn't block your commit here, though.